### ISSUE TITLE
I'm very interested in a slight variation of the use case in the description. Here is mine:

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#43136]
I'm very interested in a slight variation of the use case in the description. Here is mine:

I have an SDK that comes with a few of its own analyzers. It also selects a handful of analyzers from other analyzer packages (including those defined in-box with .NET 5 SDK) that my SDK wants to be turned on. Folks who reference *only* my package should get all the analyzers *that I selected* and no others. The fact that several analyzer packages are brought in as transitive dependencies is *by design*, but I don't want to have to infuse many hundreds of analyzers into my user's project in order to just get a few of the relevant ones that I know are particularly important/relevant to my SDK. 
But if that consuming project *also* references those other analyzer packages, or otherwise opts into seeing those rules, then they should be allowed to get the rest of the default set of analyzers from those packages.

So in other words, by virtue of an analyzer package being brought in transitively *through my package*, I should get the freedom to filter to a subset of rules to activate by default, and possibly adjust their severities. But in doing so, this should *not* disable rules that I didn't select if the consuming project has otherwise expressed interest in them.

I think we need a proper concept of recognizing whether an analyzer package was referenced transitively through a 'filtering' package to solve this problem.

_Originally posted by @AArnott in https://github.com/dotnet/roslyn/issues/42219#issuecomment-610078436_

**Comment by jinujoseph** [comment]
cc @mavasani
